### ISSUE TITLE
1.1.1d ./config does not return an error during configuration error

### ISSUE THREAD

**Issue by noloader** [openssl/openssl#10951]
I'm working from the OpenSSL 1.1.1d tarball. I'm catching a configuration error when running `./config`:

```
***** Unsupported options: no-comp --prefix=/home/jwalton/tmp/build-test --libdir=/home/jwalton/tmp/build-test/lib
```

The problem for this report is, `config` does not return an error so my script continues as if there is no error:

```
./config  no-comp --prefix=/home/jwalton/tmp/build-test --libdir=/home/jwalton/tmp/build-test/lib
Operating system: x86_64-whatever-linux2
...
***** Unsupported options: no-comp --prefix=/home/jwalton/tmp/build-test --libdir=/home/jwalton/tmp/build-test/lib
**********************
Building package
**********************
```

I should never get to the message *"Building Package"* because:

```
    KERNEL_BITS="$INSTX_BITNESS" \
./config "${CONFIG_FLAGS[*]}" \
    PKG_CONFIG_PATH=\""${BUILD_PKGCONFIG[*]}"\" \
    CPPFLAGS=\""${BUILD_CPPFLAGS[*]}" -DPEDANTIC\" \
    CFLAGS=\""${BUILD_CFLAGS[*]}"\" \
    CXXFLAGS=\""${BUILD_CXXFLAGS[*]}"\" \
    LDFLAGS=\""${BUILD_LDFLAGS[*]}"\"

if [[ "$?" -ne 0 ]]; then
    echo "Failed to configure OpenSSL"
    exit 1
fi

echo "**********************"
echo "Building package"
echo "**********************"

MAKE_FLAGS=("-j" "$INSTX_JOBS")
if ! "$MAKE" "${MAKE_FLAGS[@]}"
then
    echo "Failed to build OpenSSL"
    exit 1
fi
````

It appears `config` is returning success even during an error.

**Comment by levitte** [comment]
That's weird, 'cause we do `die` (in a perl sence) after printing that error. OK, will look into it.

**Comment by levitte** [comment]
BTW, you probably want to change this: `"${CONFIG_FLAGS[*]}"`
to: "${CONFIG_FLAGS[@]}"

The later will quote each element individually...

**Comment by levitte** [comment]
Solved, see #10953
